doc/03_reference/coverage_plan.rst

```diff
@@ -0,0 +1,299 @@
+* Interrupt raised/taken.
+
+  * Interrupt raised/taken for each available interrupt line (for wider cross coverage which interrupt is raised/taken is not relevant other than NMI vs non NMI).
+  * Interrupt raised/taken first cycle of instruction or not.
+
+* External debug request.
+* Instruction executed when debug single step enabled.
+* Instruction matches hardware trigger point.
+* Ibex operating in debug mode.
+
+PMP
+^^^
+* Each region configured with different matching modes.
+
+  * Off
+  * TOR
+  * NA4
+  * NAPOT
+
+* Each region configured with all possible permissions including locked/unlocked.
+
+  * Different permissions with MML enabled and disabled, separate cover points for R/W/X/L values with and without MML.
+
+* Access fail & pass.
+
+  * All combinations of unaligned access split across a boundary, both halves pass, neither pass, one does & one doesn’t.
+  * Higher priority entry allow access that lower priority entry prevents.
+  * Compressed instruction access (16-bit) passes PMP but 32-bit access at same address crosses PMP region boundary.
+
+* Each field of mssecfg enabled/disabled with relevant functionality tested.
+
+  * RLB - rule locking bypass.
+
+    * Modify locked region with RLB set.
+    * Try to enable RLB when RLB is disabled and locked regions present.
+
+  * MMWP - machine mode whitelist policy.
+
+    * M-mode access fail due to not matching any PMP regions.
+    * Try to disable when enabled.
+
+  * MML - machine mode lockdown policy.
+
+    * Try to disable when enabled.
+
+  * Access close to PMP region modification that allows/disallows that access.
+
+CSRs
+^^^^
+Basic read/write functionality must be tested on all implemented CSRs.
+
+* Read from CSR.
+* Write to CSR.
+
+  * Write to read only CSR.
+
+* Write illegal/unsupported value to WARL field.
+* Access to CSR disallowed due to privilege level/debug mode.
+
+CSRs addresses do not need to be crossed with the variety of CSR instructions as these all use the same basic read & write interface into ``ibex_cs_registers``.
+Coverage of the above points will be sampled at the ``ibex_cs_registers`` interface (as opposed to sampling CSR instructions).
+
+Miscellaneous
+^^^^^^^^^^^^^
+Various points of interest do not fit into the categories above.
+
+* Instruction unstalled - Cover the cycle an instruction is unstalled having just been stalled.
+* Stalled awaiting an instruction fetch.
+* Double fault.
+* Awake from sleep.
+* Interrupt/Debug whilst entering sleep.
+
+Cross Coverage
+--------------
+Much of the more complex behaviour lies at the combination of the individual microarchitectural behaviours above.
+Cross coverage is used to capture that.
+There are some broad crosses containing many bins aiming to capture all combinations of some generalised behaviours as well as some more specific ones to capture all combinations of behaviours focussed on a particular area.
```

Comment:
Nit: I think a double 's' in focused is very unusual in technical writing. A quick Google suggests that it's on the way out, even for British English. Maybe just one?